tuf/api/metadata.py

```diff
@@ -301,31 +301,31 @@ def __init__(
             raise ValueError(f'version must be < 0, got {version}')
         self.version = version
 
+    @staticmethod
+    def _common_fields_from_dict(signed_dict: Mapping[str, Any]) -> list:
+        """Returns common fields of 'Signed' instances from the passed dict
+        representation, and returns an ordered list to be passed as leading
+        positional arguments to a subclass constructor.
 
-    # Deserialization (factories).
-    @classmethod
-    def from_dict(cls, signed_dict: Mapping[str, Any]) -> 'Signed':
-        """Creates Signed object from its dict representation. """
+        See '{Root, Timestamp, Snapshot, Targets}.from_dict' methods for usage.
+
+        """
+        _type = signed_dict.pop('_type')
+        version = signed_dict.pop('version')
+        spec_version = signed_dict.pop('spec_version')
+        expires_str = signed_dict.pop('expires')
         # Convert 'expires' TUF metadata string to a datetime object, which is
         # what the constructor expects and what we store. The inverse operation
-        # is implemented in 'to_dict'.
-        signed_dict['expires'] = tuf.formats.expiry_string_to_datetime(
-                signed_dict['expires'])
-        # NOTE: We write the converted 'expires' back into 'signed_dict' above
-        # so that we can pass it to the constructor as  '**signed_dict' below,
-        # along with other fields that belong to Signed subclasses.
-        # Any 'from_dict'(-like) conversions of fields that correspond to a
-        # subclass should be performed in the 'from_dict' method of that
-        # subclass and also be written back into 'signed_dict' before calling
-        # super().from_dict.
-
-        # NOTE: cls might be a subclass of Signed, if 'from_dict' was called on
-        # that subclass (see e.g. Metadata.from_dict).
-        return cls(**signed_dict)
+        # is implemented in '_common_fields_to_dict'.
+        expires = tuf.formats.expiry_string_to_datetime(expires_str)
+        return [_type, version, spec_version, expires]
 
+    def _common_fields_to_dict(self) -> Dict[str, Any]:
```

Comment:
Can't be static because it needs `self`, but thanks for pointing out that those two are called on `super()`. I guess it's less confusing to call them on `self`, given that they are only called in the parent class. 🤷